bite-size-notes/week-6-b-free-final-time-PMP.md

```diff
@@ -0,0 +1,231 @@
+
+The method explored in particular is Pontryagin's Minimum Principle (PMP), which is computationally-efficient and can provide insights into sensitivity of the solution, but lacks global optimality guarantees unlike the more popular Hamilton-Jacobi-Bellman-Isaacs (HJBI) equation.
+
+## Scope
+
+These notes will focus on introducing free final time on a simple linear system with theory and code, explain difficulties with implementation on nonlinear and higher-order linear systems, and provide resources to explore the subject further.
+
+Learning objectives include...
+* Discovering the difference between functions and functionals.
+* Understanding the similarities of extremizing functions of a functional with finding critical points.
+* Determining necessary conditions for optimality with the Hamiltonian under Pontryagin's Minimum Principle.
+* Constructing a free final time problem of a simple linear system.
+
+The reader is expected to have prior knowledge of undergraduate-level controls and topics such as basic optimization and the Lagrangian.
+
+## Prerequisites
+
+
+### Nomenclature
+<details close>
+<summary> <em> (open to view)</em> </summary>
+<br>
+<div markdown="1">
+
+| Term            | Definition                                      |
+|-----------------|-------------------------------------------------|
+| $a$ | acceleration |
+| $H$ | Hamiltonian |
+| $J$ | cost function/functional |
+| $L$ | Lagrangian |
+| $t$ | time |
+| $T$ | terminal time |
+| $u$ | control |
+| $v$ | velocity |
+| $x$ | position in $x$-direction |
+| $x_g$ | goal position in $x$-direction |
+| $y$ | position in $y$-direction |
+| $\theta$ | angle |
+| $\lambda$ | co-state |
+| $\omega$ | angular velocity |
+| $\Psi$ | terminal cost |
+
+</div>
+</details>
+
+### Motivation
+Why do we need new math to solve free final time?
+
+For free final time problem, terminal time $T$ is no longer fixed. Thus when finding an optimal solution, the state $z$ and control $u$ are no longer the optimization variables in a cost function, but rather $z(t)$, $u(t)$ are optimization *functions* and $T$ is an optimization variable in a cost *functional*.
+
+Put another way: rather than finding extrema (minima/maxima) values for a function, we instead find extrema functions, functions that minimize or maximize a functional.
+
+### Functionals
+Functionals are mappings from a set of functions to the set of real numbers. Intuitively, one can think of them as "functions of functions", as functions are mappings from a set of numbers to another set of numbers. The calculus of variations uses small changes of functions and functionals, known as variations, to determine the extrema of functionals.
```

Comment:
The third sentence is a bit generic: connecting it with the following paragraph would help clarify; e.g. the following paragraph could answer this question

what is done in the framework of calculus of variations that is analogous to setting the derivative to zero? 

This would clarify how `The calculus of variations uses small changes of functions and functionals, known as variations, to determine the extrema of functionals`

Reply:
Good point, I've added more specificity by mentioning that it's directly analogous to the first variation being set to zero.